pulp_smash/tests/rpm/cli/test_environments.py

```diff
@@ -0,0 +1,70 @@
+# coding=utf-8
+"""Tests for RPM package environments."""
+from __future__ import unicode_literals
+
+import subprocess
+import unittest2
+
+from pulp_smash import cli, config, utils
+from pulp_smash.constants import (
+    RPM_ENV_DESC,
+    RPM_ENV_NAME,
+    RPM_FEED_URL,
+)
+
+
+class UploadPackageEnvTestCase(unittest2.TestCase):
+    """Test whether Pulp can upload package environments into a repository.
+
+    This test case covers `Pulp # 1003`_ and the corresponding Pulp Smash
+    issue `Pulp Smash #319`_.
+
+    1. Create and sync a repository.
+    2. Upload the environmrnt into this repo and check if there's any error.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create and sync a repository."""
+        cls.cfg = config.get_config()
+        cls.repo_id = utils.uuid4()
+        cls.client = cli.Client(cls.cfg)
+        cls.client.run(
+            'pulp-admin rpm repo create --repo-id {} --feed {}'
+            .format(cls.repo_id, RPM_FEED_URL).split()
+        )
+        try:
+            cls.client.run(
+                'pulp-admin rpm repo sync run --repo-id {}'
+                .format(cls.repo_id).split()
+            )
+        except subprocess.CalledProcessError:
+            cls.client.run(
+                'pulp-admin rpm repo delete --repo-id {}'
+                .format(cls.repo_id).split()
+            )
+            raise
+
+    @classmethod
+    def tearDownClass(cls):
+        """Destroy the repository."""
+        cli.Client(cls.cfg).run(
+            'pulp-admin rpm repo delete --repo-id {}'
+            .format(cls.repo_id).split()
+        )
+
+    def test_upload_environment(self):
+        """Test if package environments can be uploaded."""
+        proc_upload = cli.Client(self.cfg).machine.session().run(
+            'pulp-admin rpm repo uploads environment '
+            '--repo-id {0} --environment-id {1} '
+            '--name "{2}" --description "{3}"'
+            .format(self.repo_id, utils.uuid4(), RPM_ENV_NAME, RPM_ENV_DESC)
+        )
+        with self.subTest(comment='verify upload environments stdout'):
+            self.assertNotIn('Task Failed', proc_upload[1])
+        proc_content = cli.Client(self.cfg).run(
+            'pulp-admin rpm repo content environment --repo-id {}'
+            .format(self.repo_id).split()
+        )
+        self.assertNotEqual('\x1b[0m', proc_content.stdout)
```

Comment:
A better assertion would be checking the values passed are the ones returned by the list command:

Below is a sample output from the docs:

```
$ pulp-admin rpm repo content environment --repo-id repo_1 --match id=test-env

Description: test-env
Group Ids:   pulp_dotted_name_packages, pulp_test_packages
Id:          test-env
Name:        test-env
Options:
```

A better assertion would be checking if the name and description match what you've passed during creation.